lib/valueflow.cpp

```diff
@@ -606,7 +606,9 @@ static ValueFlow::Value truncateImplicitConversion(Token* parent, const ValueFlo
 
 static long long truncateIntValue(long long value, size_t value_size, const ValueType::Sign dst_sign)
 {
-    const MathLib::biguint unsignedMaxValue = (1ULL << (value_size * 8)) - 1ULL;
+    MathLib::biguint unsignedMaxValue = -1;
```

Comment:
this will not work well. 
if int according to the platform configuration is 16-bits then the max value should be 0xffff
but max value will be 0xffffffff if I compile cppcheck on a normal platform where int is 32 bits.